### PR TITLE
Apply state modifications once

### PR DESCRIPTION
**Pull request by StephenButtolph**
## Why this should be merged

1. This improves the current performance of the P-chain because time changes are only applied once.
2. This improves the performance of the suggested change to the P-chain because it only applies decision transactions to one state diff.

## How this works

Moves state application from verification to accept.

## How this was tested

- [X] CI

### CODE REVIEW

File: vms/platformvm/block/executor/acceptor.go [hunk 1]
```diff
@@ -172,6 +172,16 @@ func (a *acceptor) optionBlock(b, parent block.Block, blockType string) error {
 		return err
 	}
 
+	parentState, ok := a.blkIDToState[parentID]
+	if !ok {
+		return fmt.Errorf("%w %s", errMissingBlockState, parentID)
+	}
+	if parentState.onDecisionState != nil {
+		if err := parentState.onDecisionState.Apply(a.state); err != nil {
```
Comment (patrick-ogrady) [inline vms/platformvm/block/executor/acceptor.go:180]: To confirm, just calling the `onAcceptState` doesn't commit all diffs under it (for example the `wrapped decision state`)?

Comment (StephenButtolph) [reply vms/platformvm/block/executor/acceptor.go:180]: That's correct

Comment (StephenButtolph) [reply vms/platformvm/block/executor/acceptor.go:180]: We could apply the `onAcceptState` to `onDecisionState` and then apply `onDecisionState` to `a.state` if you felt that was cleaner... But this is the (slightly) more efficient way to do the same thing.

Comment (patrick-ogrady) [reply vms/platformvm/block/executor/acceptor.go:180]: I think what we have here is fine. Makes sense (should be the same thing).

**Review by patrick-ogrady — APPROVED** [review]

**Review by dhrubabasu — APPROVED** [review]

**Pull request merged**